contracts/src/deployers/aerodrome-lp/AerodromeLpHyperdriveCoreDeployer.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.22;
+
+import { IGauge } from "aerodrome/interfaces/IGauge.sol";
+import { IHyperdrive } from "../../interfaces/IHyperdrive.sol";
+import { IHyperdriveAdminController } from "../../interfaces/IHyperdriveAdminController.sol";
+import { IHyperdriveCoreDeployer } from "../../interfaces/IHyperdriveCoreDeployer.sol";
+import { AerodromeLpHyperdrive } from "../../instances/aerodrome-lp/AerodromeLpHyperdrive.sol";
+
+/// @author DELV
+/// @title AerodromeLpHyperdriveCoreDeployer
+/// @notice The core deployer for the AerodromeLpHyperdrive implementation.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+contract AerodromeLpHyperdriveCoreDeployer is IHyperdriveCoreDeployer {
+    /// @dev The Aerodrome Gauage contract. This is where the base token will be
+    ///      deposited.
+    IGauge internal immutable gauge;
```

Comment:
It seems like there is a different gauge contract for each pair of staking and rewards tokens. It would be better to encode this in extra data to avoid redeploying the deployer and deployer coordinator everytime we need to swap out the gauge contract.